### PR TITLE
Create Error File On Telegraf Failure

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
Create a file if the agent fails to start

# Description of changes
Create a file if the agent fails to start

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Tested in internal pipeline

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ var fServiceName = flag.String("service-name", "telegraf", "service name (window
 var fServiceDisplayName = flag.String("service-display-name", "Telegraf Data Collector Service", "service display name (windows only)")
 var fRunAsConsole = flag.Bool("console", false, "run as console application (windows only)")
 var fSetEnv = flag.String("setenv", "", "set an env in the configuration file in the format of KEY=VALUE")
+var fStartUpErrorFile = flag.String("errorFile", "", "file to touch if agent can't start")
```
Comment (jefchien) [inline cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go:93]: `startup-error-file` to match format of other arguments and to be clear about the purpose.

**Review by adam-mateen — DISMISSED** [review]

**Review by jefchien — APPROVED** [review]

File: cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go [hunk 2]
```diff
@@ -175,6 +176,14 @@ func reloadLoop(
 
 		err := runAgent(ctx, inputFilters, outputFilters)
 		if err != nil && err != context.Canceled {
+			if *fStartUpErrorFile != "" {
```
Comment (sky333999) [inline cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go:179]: If there is a sigterm, does it hit this code path?

Comment (sethAmazon) [reply cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go:179]: It should not since that is a context timeout

Comment (sethAmazon) [reply cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go:179]: It will also, not execute for kernal panics

**Review by Paramadon — APPROVED** [review]

**Pull request merged**